src/Components/Blazor/Build/src/Tasks/GenerateBlazorBootJson.cs

```diff
@@ -17,53 +18,103 @@ public class GenerateBlazorBootJson : Task
         public string AssemblyPath { get; set; }
 
         [Required]
-        public ITaskItem[] References { get; set; }
+        public ITaskItem[] Resources { get; set; }
+
+        [Required]
+        public bool DebugBuild { get; set; }
 
         [Required]
         public bool LinkerEnabled { get; set; }
 
+        [Required]
+        public bool CacheBootResources { get; set; }
+
         [Required]
         public string OutputPath { get; set; }
 
         public override bool Execute()
         {
-            var entryAssemblyName = AssemblyName.GetAssemblyName(AssemblyPath).Name;
-            var assemblies = References.Select(GetUriPath).OrderBy(c => c, StringComparer.Ordinal).ToArray();
-
             using var fileStream = File.Create(OutputPath);
-            WriteBootJson(fileStream, entryAssemblyName, assemblies, LinkerEnabled);
+            var entryAssemblyName = AssemblyName.GetAssemblyName(AssemblyPath).Name;
+            WriteBootJson(fileStream, entryAssemblyName);
 
             return true;
+        }
+
+        // Internal for tests
+        internal void WriteBootJson(Stream output, string entryAssemblyName)
+        {
+            var result = new BootJsonData
```

Comment:
Do we need the Blazor.Boot.json file structure to differentiate between resource types? Can we use the extension instead?
Can we bake in the expected base relative path for the resource instead of the resource name into blazor.boot.json?
 * It allows us to avoid hardcoding paths on blazor.webassembly.js
 * Won't have a meaningful impact on the the manifest size.
Can we have a simpler blazor.boot.json that looks something like:
```json
{
  "resources": {
    "_framework/wasm/dotnet.wasm": "asdfasfdadsfasfdasdf",
    "_framework/bind/App.dll": "hasdafdadsfasfda",
    ...
  },
  ...
}
```
We can build a more complex data structure on the client if we want to by doing grouping and similar things based on the extension.
For example, we can avoid including pdbs at all in the manifest if it is not a debug build and that way we don't need to special case that.

For the apps/resources etc I expect the client to simply say in the callback, these are the things I need to load and here are the paths.

Reply:
These are good points.

I would be happy to change the structure of `blazor.boot.json` and I do like the idea of not having the client hardcode knowledge of the URL formats for different resource types.

One thing that gives me pause at the moment though is this stuff I was dealing with earlier today: https://github.com/dotnet/aspnetcore/issues/18951. We're going to need some way of knowing which culture each assembly relates to, so we probably need some structure with more metadata than just (URL,hash) pairs.

Since we don't yet know how we want to structure things to provide culture info until we deal with the design questions in #18951, I'm inclined not to address this first. Would you consider it reasonable to make "optimise the `blazor.boot.json` structure" an aspect of #18951?

If you feel strongly I will change it in this PR, but it just seems a bit like duplicated work when we know it will change again.

Reply:
That's fair, although for culture sensitive stuff I would have just expected the culture to appear in the url in some way, for example: `_framework/_bin/es-ES/mscorlib.dll` and so on.

Reply:
OK, let's make a call about how we represent the culture information when we implement localisation support. We can update the `blazor.boot.json` format to whatever shape makes the most sense when we have that design.